src/zenml/zen_stores/secrets_stores/base_secrets_store.py

```diff
@@ -75,60 +106,102 @@ def __init__(
         super().__init__(**kwargs)
         self._zen_store = zen_store
 
+        self.zen_store.register_event_handler(
+            StoreEvent.WORKSPACE_DELETED, self._on_workspace_deleted
+        )
+
+        self.zen_store.register_event_handler(
+            StoreEvent.USER_DELETED, self._on_user_deleted
+        )
+
         try:
             self._initialize()
         except Exception as e:
             raise RuntimeError(
                 f"Error initializing {self.type.value} secrets store: {str(e)}"
             ) from e
 
+    @staticmethod
+    def _load_external_store_class(
+        store_config: SecretsStoreConfiguration,
+    ) -> Type["BaseSecretsStore"]:
+        """Loads the external secrets store class from the given config.
+
+        Args:
+            store_config: The configuration of the secrets store.
+
+        Returns:
+            The secrets store class corresponding to the configured external
+            secrets store.
+
+        Raises:
+            ValueError: If the configured class path cannot be imported or is
+                not a subclass of `BaseSecretsStore`.
+        """
+        # Ensured through Pydantic root validation
+        assert store_config.class_path is not None
+
+        # Import the class dynamically
+        try:
+            store_class = import_class_by_path(store_config.class_path)
```

Comment:
There is `source_utils.load_and_validate_class` which I think generalizes what you implemented here